### PR TITLE
Add consumer proguard rules to prevent minification

### LINKED ISSUE #124: `completed` event is not caught on Android on release builds
### What area is the issue related to?

Checkout Sheet Kit

### What platform does the issue affect?

Android

### What version of `@shopify/checkout-sheet-kit` are you using?

3.0.1 and 3.0.3

### Do you have reproducible example code?

_No response_

### Steps to Reproduce

- Implement `@shopify/checkout-sheet-kit`
```tsx
export const SomeComponent = () => {
  const shopifyCheckout = useShopifyCheckoutSheet();
  useEffect(() => {
    const completed = shopifyCheckout.addEventListener(
      'completed',
      (event: CheckoutCompletedEvent) => {
        console.log("I'm here");
      },
    );
    return () => {
      completed?.remove();
    };
  }, [shopifyCheckout]);

  return <SomeStuff />
};
```
- Perform a purchase on a store on Android
- The `completed` event is caught on dev builds, **not caught on release build**

### Expected Behavior

The `completed` event is caught on release build

### Actual Behavior

The `completed` event is **not** caught on release build

### Storefront domain

redacted

### Screenshots/Videos/Log output

_No response_

### PR DESCRIPTION
**Pull request by markmur**
### What changes are you making?

Adds consumer proguard rules to prevent minifcation.

Fixes #124 

---

### PR Checklist

> [!IMPORTANT]
>
> - [ ] I've added tests to support my implementation
> - [x] I have read and agree with the [Contribution Guidelines](https://github.com/shopify/checkout-sheet-kit-react-native/blob/main/.github/CONTRIBUTING.md).
> - [x] I have read and agree with the [Code of Conduct](https://github.com/shopify/checkout-sheet-kit-react-native/blob/main/.github/CODE_OF_CONDUCT.md).
> - [ ] I've updated the [README](https://github.com/shopify/checkout-sheet-kit-react-native).
> - [x] I have bumped the version number in the [`package.json` file](https://github.com/Shopify/checkout-sheet-kit-react-native/blob/main/modules/%40shopify/checkout-sheet-kit/package.json#L4).
> - [x] I have added a [Changelog](https://github.com/Shopify/checkout-sheet-kit-react-native/blob/main/CHANGELOG.md) entry.

### CODE REVIEW

File: modules/@shopify/checkout-sheet-kit/android/proguard-rules.pro [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+# Keep Checkout Sheet Kit classes
+-keep class com.shopify.checkoutsheetkit.lifecycleevents.CheckoutCompletedEvent { *; }
+-keep class com.shopify.checkoutsheetkit.pixelevents.PixelEvent { *; }
+-keep class com.shopify.checkoutsheetkit.CheckoutException { *; }
+-keep class com.shopify.checkoutsheetkit.CheckoutExpiredException { *; }
+-keep class com.shopify.checkoutsheetkit.ClientException { *; }
+-keep class com.shopify.checkoutsheetkit.HttpException { *; }
+-keep class com.shopify.checkoutsheetkit.ConfigurationException { *; }
+-keep class com.shopify.checkoutsheetkit.CheckoutSheetKitException { *; }
```
Comment (kiftio) [inline modules/@shopify/checkout-sheet-kit/android/proguard-rules.pro:12]: could/should this be added to the checkout-sheet-kit-android repo?

Comment (markmur) [reply modules/@shopify/checkout-sheet-kit/android/proguard-rules.pro:12]: Good question - do we have a rules file there already?

Comment (xseignard) [reply modules/@shopify/checkout-sheet-kit/android/proguard-rules.pro:12]: This would make sense, looking at the current rules, I deduce the issue I faced is related to the underlying android lib being minified. If so, I assume my issue would also occur on a plain android app and then it would definitely make sense to drill up the proguard rules to the android lib.

Please note I didn't need to keep the fasterxml class, but I may have another lib doing it already, I didn't check

**Review by xseignard — COMMENTED** [review]
Thanks for giving a shot

File: modules/@shopify/checkout-sheet-kit/android/proguard-rules.pro (hunk 1 above)
Comment (markmur) [reply modules/@shopify/checkout-sheet-kit/android/proguard-rules.pro:12]: @kiftio PR to add the rules to the Android kit https://github.com/Shopify/checkout-sheet-kit-android/pull/150

It's unclear to me whether we need to maintain the fasterxml class which is used in this kit. Let's ship the Android changes, upgrade the kit here and see if it works.

File: sample/ios/ReactNative/ReactNative.entitlements [hunk 2]
```diff
@@ -5,6 +5,7 @@
 	<key>com.apple.developer.associated-domains</key>
 	<array>
 		<string>webcredentials:myshopify.com</string>
+    <string>applinks:myshopify.com</string>
```
Comment (kiftio) [inline sample/ios/ReactNative/ReactNative.entitlements:8]: Is this related?

File: sample/android/app/build.gradle [hunk 3]
```diff
@@ -115,8 +110,8 @@ android {
     }
     release {
       signingConfig signingConfigs.release
-      minifyEnabled enableProguardInReleaseBuilds
-      proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
+      minifyEnabled true
+      proguardFiles "proguard-rules.pro"
```
Comment (kiftio) [inline sample/android/app/build.gradle:114]: Do we want to drop getDefaultProguardFile() part

Comment (markmur) [reply sample/android/app/build.gradle:114]: Undoing this change...

Comment (markmur) [reply sample/android/app/build.gradle]: Done

**Comment by spo-cmoore** [comment]
I'm encountering permissions issues installing the new package (or 3.1.1), is this only on my end or is it an npm publishing issue?
